envs/numpy_env.py

```diff
@@ -44,28 +45,30 @@ def step(self, action):
 
     return obs, reward, done
 
-  def _pick(self, x, y, rot):
+  def _pick(self, x, y, z, rot):
     ''''''
     if self._isHolding():
       return self.held_object
 
     height_sorted_objects = sorted(self.objects, key=lambda x: x.pos[-1], reverse=True)
     for obj in height_sorted_objects:
-      if obj.isGraspValid([x,y], rot):
+      if obj.isGraspValid([x,y,z], rot):
         obj.removeFromHeightmap(self.heightmap)
         return obj
 
     return None
 
-  def _place(self, x, y, rot):
+  def _place(self, x, y, z, rot):
     ''''''
     pass
 
   def _getObservation(self):
     ''''''
+    for o in self.objects:
+      o.addToHeightmap(self.heightmap)
```

Comment:
Does this re-add objects to the heightmap multiple times? Seems a bit weird unless we are reseting the height map to zeros at every time step.

Reply:
Yes, I fixed this